src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Areas/Identity/Pages/Account/ExternalLoginTests.cs

```diff
@@ -0,0 +1,576 @@
+            VerifyAllMocks(signInManager);
+            
+            AssertRedirectedToLoginPageUponFailure(result, loginService);
+        }
+
+        [Fact]
+        public async Task Login_EmailClaimNotAvailable()
+        {
+            var claimsPrincipal = AuthorizationHandlersTestUtil.CreateClaimsPrincipal(
+                Guid.NewGuid(),
+                new Claim(ClaimTypes.GivenName, "FirstName"),
+                new Claim(ClaimTypes.Surname, "LastName"));
+            
+            var signInManager = new Mock<ISignInManagerDelegate>(Strict);
+
+            signInManager
+                .Setup(s => s.GetExternalLoginInfoAsync(null))
+                .ReturnsAsync(new ExternalLoginInfo(
+                    claimsPrincipal,
+                    LoginProvider,
+                    ProviderKey,
+                    LoginProviderDisplayName));
+
+            await using var usersDbContext = InMemoryUserAndRolesDbContext();
+            await using var contentDbContext = InMemoryApplicationDbContext();
+            
+            var loginService = BuildService(
+                signInManager.Object,
+                usersDbContext: usersDbContext,
+                contentDbContext: contentDbContext);
+
+            var result = await loginService.OnGetCallbackAsync(ReturnUrl);
+            VerifyAllMocks(signInManager);
+            
+            AssertRedirectedToLoginPageUponFailure(result, loginService);
+        }
+
+        [Fact]
+        public async Task Login_UninvitedNewUser_Failure()
+        {
+            var claimsPrincipal = CreateClaimsPrincipal(
+                "uninviteduser@example.com",
+                "FirstName",
+                "LastName");
+
+            var userManager = new Mock<IUserManagerDelegate>(Strict);
+            var signInManager = new Mock<ISignInManagerDelegate>(Strict);
+
+            await using var usersDbContext = InMemoryUserAndRolesDbContext();
+            await using var contentDbContext = InMemoryApplicationDbContext();
+            
+            var loginService = BuildService(
+                signInManager.Object,
+                userManager.Object,
+                contentDbContext,
+                usersDbContext);
+
+            signInManager
+                .Setup(s => s.GetExternalLoginInfoAsync(null))
+                .ReturnsAsync(new ExternalLoginInfo(
+                    claimsPrincipal,
+                    LoginProvider,
+                    ProviderKey,
+                    LoginProviderDisplayName));
+
+            var result = await loginService.OnGetCallbackAsync(ReturnUrl);
+            VerifyAllMocks(userManager, signInManager);
+                
+            AssertRedirectedToLoginPageUponFailure(result, loginService);
+        }
+
+        [Fact]
+        public async Task Login_InvitedUser_Success()
+        {
+            var contextId = Guid.NewGuid().ToString();
+
+            var email = "inviteduser@example.com";
```

Comment:
Could use `const string` rather than `var` here and in a few other places throughout these tests to indicate these variables don't get reassigned.

Reply:
cool - done thanks!